src/Symfony/Component/Mailer/Transport.php

```diff
@@ -54,15 +56,33 @@ class Transport
 
     private $factories;
 
-    public static function fromDsn(string $dsn, EventDispatcherInterface $dispatcher = null, HttpClientInterface $client = null, LoggerInterface $logger = null): TransportInterface
+    /**
+     * @param EventDispatcherInterface|null $dispatcher
+     * @param HttpClientInterface|null      $client
+     * @param LoggerInterface|null          $logger
+     */
+    public static function fromDsn(string $dsn/*, EventDispatcherInterface $dispatcher = null, HttpClientInterface $client = null, LoggerInterface $logger = null*/): TransportInterface
     {
+        $dispatcher = 2 <= \func_num_args() ? func_get_arg(1) : null;
```

Comment:
Please don't see this as a review ;-) I was just wondering what the added value of this change was? Not to complain; but just to understand and learn. I've not seen this before, and it seems like it would do exactly the same as the previous implementation. But maybe I'm oblivious to some `func_*_args()` magic :-)

Thanks.

Reply:
The transport class is neither internal nor final. This is why I have to assume that someone might have created a child class overriding the methods I've changed.

I wanted to widen the parameter type for the dispatcher parameter. However, that would break the child class. This is why I removed the optional parameters instead and fetch them via those `func_*()` functions.

Reply:
Ahh I see, because they would have implemented the Symfony `EventDispatcherInterface` as a parameter type-hint. I did not know you could do this. I actually thought this was a breaking change, but apparently not. Thanks for sharing! Always nice to learn new tricks.